frontends/web/src/components/transactions/transaction.tsx

```diff
@@ -146,25 +151,47 @@ const Status = ({
 
 type TAmountsProps = {
   amount: IAmount;
+  fee: IAmount;
   type: TTransactionType;
 }
 
 const Amounts = ({
   amount,
+  fee,
   type,
 }: TAmountsProps) => {
   const { defaultCurrency } = useContext(RatesContext);
   const conversion = amount?.conversions && amount?.conversions[defaultCurrency];
   const sign = getTxSign(type);
   const txTypeClass = `txAmount-${type}`;
   const conversionPrefix = amount.estimated ? '\u2248' : null; // ≈
+
+  // const [fee, setFee] = useState<IAmount>();
+
+  // useEffect(() => {
+  //   if (
+  //     type === 'send_to_self'
+  //     && !fee?.amount
+  //     && accountCode && internalID
+  //   ) {
+  //     getTransaction(accountCode, internalID).then(transaction => {
+  //       setFee(transaction?.fee);
+  //     });
+  //   }
+  // }, [accountCode, fee, internalID, type]);
+
+  const sendToSelf = type === 'send_to_self';
+  const coinAmount = sendToSelf ? fee.amount : amount.amount;
```

Comment:
this should work for coins, but for tokens, we'd need fee.unit, as when a erc20-USDT is sent to self (for whatever reason), the fee is in ETH .. or gas?